orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -144,13 +147,34 @@ def _forward(self, *inputs, **kwargs):
             if build_gradient_graph:
                 self._current_input_shape = new_input_shape
                 self._build_training_graph()
-                self._create_training_session()
+                # Create execution session creates both the training_session as well
+                # as the inference_session. The inference_session is only created when
+                # the model is set to eval mode. However, the training session is always
+                # created.
+                self._create_execution_session()
```

Comment:
Nit: Not introduced in this PR, but I wonder if we can consolidate the calls to `self._create_execution_session()` down to a single call site. Looking at the code, I'm not sure if we'd ever call `self._create_execution_session()` more than once during a single call to `forward()`.